File: SiriusQuality/SiriusQuality-Penman/src/cpp/SiriusQuality_Penman/PotentialET_PenmanComponent.cpp
#include "PotentialET_PenmanComponent.h"
using namespace SiriusQuality-Penman;
PotentialET_PenmanComponent::PotentialET_PenmanComponent()
{
       
}


double PotentialET_PenmanComponent::getlambdaV(){ return this->lambdaV; }
double PotentialET_PenmanComponent::getpsychrometricConstant(){ return this->psychrometricConstant; }
double PotentialET_PenmanComponent::getAlpha(){ return this->Alpha; }
int PotentialET_PenmanComponent::getih(){ return this->ih; }
double PotentialET_PenmanComponent::getspecificHeatCapacityAir(){ return this->specificHeatCapacityAir; }
double PotentialET_PenmanComponent::getrhoDensityAir(){ return this->rhoDensityAir; }

void PotentialET_PenmanComponent::setlambdaV(double _lambdaV)
{
    _PriestlyTaylor.setlambdaV(_lambdaV);
    _Penman.setlambdaV(_lambdaV);
}
void PotentialET_PenmanComponent::setpsychrometricConstant(double _psychrometricConstant)
{
    _PriestlyTaylor.setpsychrometricConstant(_psychrometricConstant);
    _Penman.setpsychrometricConstant(_psychrometricConstant);
}
void PotentialET_PenmanComponent::setAlpha(double _Alpha)
{
    _PriestlyTaylor.setAlpha(_Alpha);
    _Penman.setAlpha(_Alpha);
}
void PotentialET_PenmanComponent::setih(int _ih)
{
    _PriestlyTaylor.setih(_ih);
}
void PotentialET_PenmanComponent::setspecificHeatCapacityAir(double _specificHeatCapacityAir)
{
    _Penman.setspecificHeatCapacityAir(_specificHeatCapacityAir);
}
void PotentialET_PenmanComponent::setrhoDensityAir(double _rhoDensityAir)
{
    _Penman.setrhoDensityAir(_rhoDensityAir);
}
void PotentialET_PenmanComponent::Calculate_Model(PotentialET_PenmanState &s, PotentialET_PenmanState &s1, PotentialET_PenmanRate &r, PotentialET_PenmanAuxiliary &a, PotentialET_PenmanExogenous &ex)
{
    _PriestlyTaylor.Calculate_Model(s, s1, r, a, ex);
    _Penman.Calculate_Model(s, s1, r, a, ex);
}
PotentialET_PenmanComponent::PotentialET_PenmanComponent(PotentialET_PenmanComponent& toCopy)
{
    lambdaV = toCopy.getlambdaV();
    psychrometricConstant = toCopy.getpsychrometricConstant();
    Alpha = toCopy.getAlpha();
    ih = toCopy.getih();
    specificHeatCapacityAir = toCopy.getspecificHeatCapacityAir();
    rhoDensityAir = toCopy.getrhoDensityAir();
}

File: SiriusQuality/SiriusQuality-Penman/src/java/SiriusQuality_Penman/PotentialET_PenmanComponent.java
public class PotentialET_PenmanComponent
{
    
    public PotentialET_PenmanComponent() { }

    PriestlyTaylor _PriestlyTaylor = new PriestlyTaylor();
    Penman _Penman = new Penman();

    public double getlambdaV()
    { return _PriestlyTaylor.getlambdaV(); }
    public void setlambdaV(double _lambdaV){
    _PriestlyTaylor.setlambdaV(_lambdaV);
    _Penman.setlambdaV(_lambdaV);
    }

    public double getpsychrometricConstant()
    { return _PriestlyTaylor.getpsychrometricConstant(); }
    public void setpsychrometricConstant(double _psychrometricConstant){
    _PriestlyTaylor.setpsychrometricConstant(_psychrometricConstant);
    _Penman.setpsychrometricConstant(_psychrometricConstant);
    }

    public double getAlpha()
    { return _PriestlyTaylor.getAlpha(); }
    public void setAlpha(double _Alpha){
    _PriestlyTaylor.setAlpha(_Alpha);
    _Penman.setAlpha(_Alpha);
    }

    public Integer getih()
    { return _PriestlyTaylor.getih(); }
    public void setih(Integer _ih){
    _PriestlyTaylor.setih(_ih);
    }

    public double getspecificHeatCapacityAir()
    { return _Penman.getspecificHeatCapacityAir(); }
    public void setspecificHeatCapacityAir(double _specificHeatCapacityAir){
    _Penman.setspecificHeatCapacityAir(_specificHeatCapacityAir);
    }

    public double getrhoDensityAir()
    { return _Penman.getrhoDensityAir(); }
    public void setrhoDensityAir(double _rhoDensityAir){
    _Penman.setrhoDensityAir(_rhoDensityAir);
    }
    public void  Calculate_Model(PotentialET_PenmanState s, PotentialET_PenmanState s1, PotentialET_PenmanRate r, PotentialET_PenmanAuxiliary a, PotentialET_PenmanExogenous ex)
    {
        _PriestlyTaylor.Calculate_Model(s, s1, r, a, ex);
        _Penman.Calculate_Model(s, s1, r, a, ex);
    }
    private double lambdaV;
    private double psychrometricConstant;
    private double Alpha;
    private Integer ih;
    private double specificHeatCapacityAir;
    private double rhoDensityAir;
    public PotentialET_PenmanComponent(PotentialET_PenmanComponent toCopy) // copy constructor 
    {
        this.lambdaV = toCopy.getlambdaV();
        this.psychrometricConstant = toCopy.getpsychrometricConstant();
        this.Alpha = toCopy.getAlpha();
        this.ih = toCopy.getih();
        this.specificHeatCapacityAir = toCopy.getspecificHeatCapacityAir();
        this.rhoDensityAir = toCopy.getrhoDensityAir();

    }
}